### DISCUSSION TITLE
How add a theia plug-in

### DISCUSSION THREAD

**Discussion by krishnagogada** [eclipse-theia/theia#9067] [Q&A] [answered]
I take a sample example todo-highligther-plugin from [github.com/eclipse/che-theia-samples/tree/master/samples](https://github.com/eclipse/che-theia-samples/tree/master/samples).

I build and publish the package converting into VSIX.

I download the todo-highligther-plugin in plugins folder.

![image](https://user-images.githubusercontent.com/54623105/107842643-df3ca300-6dea-11eb-8cfe-a1c275634377.png)

From output it shows downloaded plugin also. But I got this error

![image](https://user-images.githubusercontent.com/54623105/107842687-2460d500-6deb-11eb-9046-3e558122197d.png)

and also I follow the steps as given here [theia-ide.org/docs/authoring_plugins](https://theia-ide.org/docs/authoring_plugins), but I didn't get hosted mode, so I added `@theia/plugin-ext`. Then output is stays at loading. When remove `@theia/plugin-ext` then I get output.

Did I miss any steps? Can you help me out?
Thank you in advance.

**Comment by vince-fugnitto** [comment, marked as answer]
@krishnagogada I verified with the [todo-highlighter-plugin](https://github.com/eclipse/che-theia-samples/tree/master/samples/todo-highlighter-plugin) and it worked correctly using this repository's examples (please note that `theia-plugins` only work in the browser target at the moment).

<img width="500" alt="Screen Shot 2021-02-13 at 1 52 39 PM" src="https://user-images.githubusercontent.com/40359487/107858645-14ed8600-6e03-11eb-9e0b-2097348648cf.png">

I did the following:
1. `git clone git@github.com:eclipse/che-theia-samples.git`
2. `cd che-theia-samples/samples/todo-highlighter-plugin`
3. `yarn`
4. copied `todo_highlighter_plugin.theia` under the `plugins` folder
5. start the example-browser application
6. performed the example as documented `// TODO`

**Reply by krishnagogada** [reply]
Thanks @vince-fugnitto i don't know that to copy `todo_highlighter_plugin.theia` and paste in `plugins` folder. I got the output.
And i followed these instructions [github.com/eclipse-theia/theia/blob/master/doc/Developing.md#quick-start](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#quick-start) then i got `Hosted Mode` option also.
